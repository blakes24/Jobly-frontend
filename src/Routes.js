import { useState } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

const Routes = () => {
	return (
		<>
			<Navbar />
			<Switch>
				<Route exact path="/">
					<Home />
				</Route>
				<Route exact path="/companies">
					<List />
				</Route>
				<Route exact path="/companies/:handle">
					<CompanyDetail />
				</Route>
				<Route exact path="/jobs">
					<List />
				</Route>
				<Route exact path="/login">
					<LoginForm />
				</Route>
				<Route exact path="/signup">
					<SignupForm />
				</Route>
				<Route exact path="/profile">
					<Profile />
				</Route>
				<Route>
					<p>Page not found.</p>
				</Route>
			</Switch>
		</>
	);
};

export default Routes;
