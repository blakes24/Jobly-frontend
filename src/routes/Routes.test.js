import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Routes from './Routes';
import JoblyApi from '../helpers/api';

jest.mock('../helpers/api');
jest.mock('jwt-decode', () => () => testUser);

afterEach(() => {
  jest.resetAllMocks();
});

const token = process.env.REACT_APP_TOKEN;

const testUser = {
  username     : 'testuser',
  firstName    : 'Test',
  lastName     : 'User',
  email        : 'joel@joelburton.com',
  isAdmin      : false,
  applications : []
};

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <Routes />
    </MemoryRouter>
  );
});

it('matches snapshot', function() {
  const { asFragment } = render(
    <MemoryRouter>
      <Routes />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('displays homepage', function() {
  const { getByText } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  expect(getByText(`The best place to find your next job.`)).toBeInTheDocument();
});

it('redirects protected route to home if user is not logged in', function() {
  const { getByText } = render(
    <MemoryRouter initialEntries={[ '/jobs' ]}>
      <Routes />
    </MemoryRouter>
  );
  expect(getByText(`The best place to find your next job.`)).toBeInTheDocument();
});

it('lets a user log in', async function() {
  JoblyApi.login.mockResolvedValue(token);
  JoblyApi.getUser.mockResolvedValue(testUser);

  const { getByText, getByLabelText, findByText, getByRole } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  fireEvent.click(getByText('Log In'));

  expect(getByLabelText(`Username`)).toBeInTheDocument();

  fireEvent.input(getByLabelText('Username'), {
    target : { value: 'testuser' }
  });
  fireEvent.input(getByLabelText('Password'), {
    target : { value: 'password' }
  });
  fireEvent.submit(getByRole('button', { name: 'Log In' }));

  expect(await findByText('Welcome back testuser!')).toBeInTheDocument();
});

it('displays signup form when nav link is clicked', function() {
  const { getByText, getByLabelText } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  fireEvent.click(getByText('Sign Up'));

  expect(getByLabelText(`First Name`)).toBeInTheDocument();
});

it('shows jobs page when logged in', async function() {
  JoblyApi.login.mockResolvedValue(token);
  JoblyApi.getUser.mockResolvedValue(testUser);
  JoblyApi.getJobs.mockResolvedValue([]);

  const { getByText, getByLabelText, findByText, getByRole } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  fireEvent.click(getByText('Log In'));

  expect(getByLabelText(`Username`)).toBeInTheDocument();

  fireEvent.input(getByLabelText('Username'), {
    target : { value: 'testuser' }
  });
  fireEvent.input(getByLabelText('Password'), {
    target : { value: 'password' }
  });
  fireEvent.submit(getByRole('button', { name: 'Log In' }));

  expect(await findByText('Welcome back testuser!')).toBeInTheDocument();

  fireEvent.click(getByText('Jobs'));

  expect(getByText('Submit')).toBeInTheDocument();
});

it('shows companies page when logged in', async function() {
  JoblyApi.login.mockResolvedValue(token);
  JoblyApi.getUser.mockResolvedValue(testUser);
  JoblyApi.getCompanies.mockResolvedValue([]);

  const { getByText, getByLabelText, findByText, getByRole } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  fireEvent.click(getByText('Log In'));

  expect(getByLabelText(`Username`)).toBeInTheDocument();

  fireEvent.input(getByLabelText('Username'), {
    target : { value: 'testuser' }
  });
  fireEvent.input(getByLabelText('Password'), {
    target : { value: 'password' }
  });
  fireEvent.submit(getByRole('button', { name: 'Log In' }));

  expect(await findByText('Welcome back testuser!')).toBeInTheDocument();

  fireEvent.click(getByText('Companies'));

  expect(getByText('Submit')).toBeInTheDocument();
});

it('shows profile page when logged in', async function() {
  JoblyApi.login.mockResolvedValue(token);
  JoblyApi.getUser.mockResolvedValue(testUser);

  const { getByText, getByLabelText, findByText, getByRole } = render(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Routes />
    </MemoryRouter>
  );
  fireEvent.click(getByText('Log In'));

  expect(getByLabelText(`Username`)).toBeInTheDocument();

  fireEvent.input(getByLabelText('Username'), {
    target : { value: 'testuser' }
  });
  fireEvent.input(getByLabelText('Password'), {
    target : { value: 'password' }
  });
  fireEvent.submit(getByRole('button', { name: 'Log In' }));

  expect(await findByText('Welcome back testuser!')).toBeInTheDocument();

  fireEvent.click(getByText('Profile'));

  expect(getByText('Update Profile')).toBeInTheDocument();
});
