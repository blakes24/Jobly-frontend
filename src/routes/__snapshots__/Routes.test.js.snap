// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar navbar navbar-expand-md navbar-dark"
  >
    <a
      aria-current="page"
      class="navbar-brand active"
      eventkey="1"
      href="/"
    >
      Jobly
    </a>
    <button
      aria-controls="navbar-collapse"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="justify-content-end navbar-collapse collapse"
      id="navbar-collapse"
    >
      <div
        class="ml-auto navbar-nav"
      >
        <a
          class="nav-link"
          data-rb-event-key="5"
          href="/login"
        >
          Log In
        </a>
        <a
          class="nav-link"
          data-rb-event-key="6"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="Home"
    >
      <h1>
        Jobly
      </h1>
      <p>
        The best place to find your next job.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
